diagnostic-settings.tf

```diff
@@ -6,6 +6,16 @@ resource "azurerm_monitor_diagnostic_setting" "kv-ds" {
   enabled_log {
     category = "AuditEvent"
   }
+
+  metric {
+    category = "AllMetrics"
+    enabled  = false
+
+    retention_policy {
+      days    = 0
+      enabled = false
+    }
```

Comment:
retention policy is deprecated isn't it?

I don't think there's a policy touching this.

Reply:
Yeah it is deprecated - I've switched to to a ignore_changes instead.

You are correct it looks like its a miss-match in the default values in the Azure terraform provider and the default values on the actual KeyVault resource in Azure.